homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,281 @@
+                        global_options=self._global_options, devices=devices
+                    )
+
+                    return self.async_create_entry(title="", data={})
+            if CONF_DEVICE_ID in user_input:
+                self._selected_device_event_code = user_input[CONF_DEVICE_ID]
+                self._selected_device = {}
+                return await self.async_step_set_device_options()
+
+            if not errors:
+                self.update_config_data(global_options=self._global_options)
+
+                return self.async_create_entry(title="", data={})
+
+        device_registry = await async_get_registry(self.hass)
+        device_entries = async_entries_for_config_entry(
+            device_registry, self._config_entry.entry_id
+        )
+        self._device_entries = device_entries
+
+        devices = {
+            entry.id: entry.name_by_user if entry.name_by_user else entry.name
+            for entry in device_entries
+        }
+
+        options = {
+            vol.Optional(CONF_DEBUG, default=self._config_entry.data[CONF_DEBUG]): bool,
+            vol.Optional(
+                CONF_AUTOMATIC_ADD, default=self._config_entry.data[CONF_AUTOMATIC_ADD],
+            ): bool,
+            vol.Optional(CONF_DEVICE_ID): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._rfxobj.device, data_bits=user_input.get(CONF_DATA_BITS)
+            )
+            try:
+                command_on = (
+                    int(user_input.get(CONF_COMMAND_ON), 16)
+                    if user_input.get(CONF_COMMAND_ON)
+                    else None
+                )
+                command_off = (
+                    int(user_input.get(CONF_COMMAND_OFF), 16)
+                    if user_input.get(CONF_COMMAND_OFF)
+                    else None
+                )
+            except (NameError, ValueError):
+                errors = {"base": "invalid_input_2262"}
+
+            if not errors:
+                devices = {}
+                devices[self._selected_device_event_code] = {
+                    CONF_DEVICE_ID: device_id,
+                    CONF_FIRE_EVENT: user_input.get(CONF_FIRE_EVENT, False),
+                    CONF_SIGNAL_REPETITIONS: user_input.get(CONF_SIGNAL_REPETITIONS, 1),
+                }
+
+                device = devices[self._selected_device_event_code]
+                if (
+                    user_input.get(CONF_OFF_DELAY_ENABLED)
+                    and user_input.get(CONF_OFF_DELAY) is not None
+                ):
```

Comment:
Why do we need delay_enabled?

Reply:
If off_delay must be disabled, it must be None. A value of "0" for example still means it is enabled and used. It seemed cleaner to give the user the option to use it and define value in separate input items.

Alternatively we could define behavior as disable means remove the input from the field, but I have to check if that can even work, not sure about that.

Reply:
@elupus I'm not getting it to work without. If I remove the default "0"
`vol.Optional(CONF_OFF_DELAY, default=device_data.get(CONF_OFF_DELAY)): int,`
And have a device which has off_delay, if I remove the input from the form it says "user input malformed" (even though it's `vol.Optional`. If I leave in the 0 and then remove the value it accepts, but the returned data contains the key with value '0'. I have no options to distinguish between someone wanting `off_delay` of `0` and someone wanting to disable `off_delay`

I think we need this delay_enabled.

Reply:
Can we use suggested value instead of default?

https://developers.home-assistant.io/docs/data_entry_flow_index#show-form

Reply:
I tried it before, it's not working because when `off_delay` is configured and when you set the suggested value and then clear the field in the form, the `user_input` contains the field with `0` value. That wouldn't give me a way to distinguish between `0` and disabled.

It did got me thinking about an alternative strategy, what seems to work is not to define the field as `int`, but as `str`. When using the suggested value and clearing the field, the `user_input` does not contain the field. The downside is that users can enter any character in the field and will only get a parsing error after submitting, but it could work. It would need some logic, because e.g. code below gives a string "None" in the field, so we only should set suggested value when there is an off_delay.

```
vol.Optional(
  CONF_OFF_DELAY,
  description={
    "suggested_value": str(device_data.get(CONF_OFF_DELAY))
  },
```

Reply:
Resolving this, removed `off_delay_enabled` and went for the `str` solution. Has a small setback that users can enter wrong data, but they get a clear message.